### PR TITLE
bpo-36588: For AIX remove major version from sys.platform

### PR DESCRIPTION
**Pull request by aixtools**
https://bugs.python.org/issue36588

### CODE REVIEW

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -1042,6 +1043,13 @@ always available.
       older Python versions include the version number, it is recommended to
       always use the ``startswith`` idiom presented above.
 
+   .. versionchanged:: 3.8
+      On AIX, :attr:`sys.platform` doesn't contain the major version anymore.
+      already uses the ``startswith`` idiom discussed above no changes to
+      existing code is expected. Remember, this is the version of AIX Python3
+      was built on, not running on. To get the version you are running on,
+      e.g., 6 or 7, use ``os.uname()[3]``
```
Comment (vstinner) [inline Doc/library/sys.rst]: Hum, this explanation is quite complicated. I would suggest to modify the idiom above to add AIX:

```

      if sys.platform.startswith('freebsd'):
          # FreeBSD-specific code here...
      elif sys.platform.startswith('linux'):
          # Linux-specific code here...
      elif sys.platform.startswith('aix'):
          # AIX-specific code here...
```

and copy/paste the Linux change:

```
      On AIX, :attr:`sys.platform` doesn't contain the major version anymore.
      It is always ``'aix'``, instead of ``'aix3'`` .. ``'aix7'``.  Since
      older Python versions include the version number, it is recommended to
      always use the ``startswith`` idiom presented above.
```

File: Doc/whatsnew/3.8.rst [hunk 2]
```diff
@@ -679,6 +679,16 @@ Porting to Python 3.8
...
+Porting Python code
+-------------------
+  XXX :issue:`36588`: On AIX, sys.platform doesn't contain the major version
+  anymore. It is now always 'aix', instead of 'aix3', 'aix4', ..., 'aix7'
+  depending on the AIX version used to build Python.
+  Replace sys.platform == 'aixX' with sys.platform == 'aix' or
+  with sys.platform.startswith('aix').
+  Code using sys.platform.startswith('aix') will work without any change.
+
+  (Contributed by M. Felt)
```
Comment (vstinner) [inline Doc/whatsnew/3.8.rst]: Please move this change into the "Changes in the Python API" section above.

I suggest to reuse the same short text:

```
* On AIX, :attr:`sys.platform` doesn't contain the major version anymore.
   It is always ``'aix'``, instead of ``'aix3'`` .. ``'aix7'``.  Since
   older Python versions include the version number, it is recommended to
   always use the ``sys.platform.startswith('aix')``. (Contributed by M. Felt in :issue:`36588`:.)
```

File: Misc/NEWS.d/next/Core and Builtins/2019-04-11-14-36-55.bpo-36588.wejLoC.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+On AIX, remove the major version in sys.platform
+It is now always 'aix', instead of 'aix3', 'aix4', ..., 'aix7'
+that reflected the AIX version used to build Python.
+Contributed by M. Felt
```
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2019-04-11-14-36-55.bpo-36588.wejLoC.rst]: Again, basic copy/paste again the same text.

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/library/sys.rst (hunk 1 above)
Comment (aixtools) [reply Doc/library/sys.rst]: OK. I was tempted to just copy/paste/substitute but the situation with AIX is not the same as it was with linux in 2011 as linux3 was becoming popular and there was a lot of code using sys.platform == 'linux2' (at least I read that 'between the lines').

The current practice is to use sys.platform.startswith('aix') because it is almost certain the number would not be correct. So, I felt silly 'recommending' something that is already common practice.

The value of this change is for the future - as it was nearly 7 years ago for linux.

If I do a search of all the code I see a mixture of sys.platform == 'linux' and sys.platform.startswith('linux') (and still one in test_util.py that sets sys.platform = 'linux2' (setting it, not testing it).

When I do the same for aix it always:
`find . | xargs grep sys.platform | grep aix`
it is either sys.platform.startswith("aix"), sys.platform.startswith('aix'), or  sys.platform[:3] == 'aix'.

As to AIX - this is the same for the (limited?) code review I have done of programs that have AIX specific sections.

Looking forward - code will be able to use sys.platform == 'aix'.

As to being 'complex' - my apologies. I tend to be verbose.

In any case, I'll make the changes as you request - but will also do them again, should that be needed. Getting the right balance in documentation is not simple.

**Comment by aixtools** [comment]
I have made the requested changes; please review again.

File: Doc/whatsnew/3.8.rst [hunk 4]
```diff
@@ -679,7 +679,6 @@ Porting to Python 3.8
 This section lists previously described changes and other bugfixes
 that may require changes to your code.
 
-
```
Comment (vstinner) [inline Doc/whatsnew/3.8.rst]: Pleave revert this unrelated change.

**Review by vstinner — APPROVED** [review]
LGTM. Just revert the newline change.

File: Doc/whatsnew/3.8.rst [hunk 5]
```diff
@@ -700,6 +699,11 @@ Changes in Python behavior
   raised when getting the attribute from the type dictionary are no longer
   ignored.  (Contributed by Serhiy Storchaka in :issue:`35459`.)
 
+* On AIX, :attr:`sys.platform` doesn't contain the major version anymore.
+  It is always ``'aix'``, instead of ``'aix3'`` .. ``'aix7'``.  Since
+  older Python versions include the version number, it is recommended to
+  always use the ``sys.platform.startswith('aix')``.
+  (Contributed by M. Felt in :issue:`36588`:.)
```
Comment (ZackerySpytz) [inline Doc/whatsnew/3.8.rst]:
```suggestion
  (Contributed by M. Felt in :issue:`36588`.)
```

The colon just before the period is unneeded.

File: Misc/NEWS.d/next/Core and Builtins/2019-04-11-14-36-55.bpo-36588.wejLoC.rst [hunk 6]
```diff
@@ -0,0 +1,5 @@
+On AIX, :attr:`sys.platform` doesn't contain the major version anymore.
+It is always ``'aix'``, instead of ``'aix3'`` .. ``'aix7'``.  Since
+older Python versions include the version number, it is recommended to
+always use the ``sys.platform.startswith('aix')``.
+Contributed by M. Felt in :issue:`36588`:.
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Core and Builtins/2019-04-11-14-36-55.bpo-36588.wejLoC.rst]: There's no need to include the issue number in the text of a news entry (this is mentioned explicitly in the UI of `blurb add`). The issue number is extracted from the filename and prepended to the text (see also https://pypi.org/project/blurb/).

So, I would suggest the following:
```suggestion
Contributed by M. Felt.
```

**Comment by vstinner** [comment]
Thanks @aixtools: that's what I expected when I wrote "properly document the change" ;-)
